specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2020-04-01/cosmos-db.json

```diff
@@ -0,0 +1,7517 @@
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB SQL storedProcedure"
+        },
+        "body": {
+          "type": "string",
+          "description": "Body of the Stored Procedure"
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "SqlUserDefinedFunctionResource": {
+      "type": "object",
+      "description": "Cosmos DB SQL userDefinedFunction resource object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB SQL userDefinedFunction"
+        },
+        "body": {
+          "type": "string",
+          "description": "Body of the User Defined Function"
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "SqlTriggerResource": {
+      "type": "object",
+      "description": "Cosmos DB SQL trigger resource object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB SQL trigger"
+        },
+        "body": {
+          "type": "string",
+          "description": "Body of the Trigger"
+        },
+        "triggerType": {
+          "type": "string",
+          "enum": [
+            "Pre",
+            "Post"
+          ],
+          "description": "Type of the Trigger",
+          "x-ms-enum": {
+            "name": "triggerType",
+            "modelAsString": true
+          }
+        },
+        "triggerOperation": {
+          "type": "string",
+          "enum": [
+            "All",
+            "Create",
+            "Update",
+            "Delete",
+            "Replace"
+          ],
+          "description": "The operation the trigger is associated with",
+          "x-ms-enum": {
+            "name": "triggerOperation",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "MongoDBDatabaseResource": {
+      "type": "object",
+      "description": "Cosmos DB MongoDB database resource object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB MongoDB database"
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "MongoDBCollectionResource": {
+      "type": "object",
+      "description": "Cosmos DB MongoDB collection resource object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB MongoDB collection"
+        },
+        "shardKey": {
+          "description": "A key-value pair of shard keys to be applied for the request.",
+          "$ref": "#/definitions/ShardKeys"
+        },
+        "indexes": {
+          "description": "List of index keys",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MongoIndex"
+          }
+        },
+        "analyticalStorageTimeToLive": {
```

Comment:
I believe these should all be "analyticalStorageTtl". In Constants.cs, the string value of the property differs from the name.